README.md

```diff
@@ -40,8 +61,16 @@ end
 ```
 
 - What is mystery2(123)?
+mystery2(123) = (3) + mystery2(12) = 5
+m2(12) = 1 + m2(1) = 2
+m2(1) = 1
```

Comment:
:) The logic is correct. but m2(12) should be 2 + m2(1) which changes the rest of the recursion math.